pkg/ingestor/parser/common/license.go

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"hash/fnv"
+	"strings"
+
+	"github.com/guacsec/guac/pkg/assembler/clients/generated"
+	"golang.org/x/exp/slices"
+)
+
+var ignore = []string{
+	"AND",
+	"OR",
+	"WITH",
+}
+
+// Could add exceptions to ignore list, as they are not licenses:
```

Comment:
Is there a way to query OSI approved licenses and only use those? Do we want to do that?

Reply:
The plan is to stick to the SPDX license list, which both SPDX and CDX use: https://spdx.org/licenses/
It includes both OSI approved and some non OSI approved licenses, reasoning here: https://github.com/spdx/license-list-XML/blob/main/DOCS/license-inclusion-principles.md

Reply:
A license expression can include both `license-id` which points to an actual license, or also `license-exception` (https://spdx.github.io/spdx-spec/v2.3/SPDX-license-expressions/#d44-exception-with-operator). I copied the exception list here, as we maybe don't want to create GUAC Licnese nodes for exceptions, as they are not licenses. But maybe GUAC users want to easily query for packages with a particular exception, so that would be a reason to not ignore them.